Makefile

```diff
@@ -1,5 +1,12 @@
 VERSION := v0.1.0
 
+ORG_PATH := github.com/Azure
+PROJECT_NAME := eraser
+REPO_PATH := $(ORG_PATH)/$(PROJECT_NAME)
+BUILD_COMMIT := $(shell git rev-parse --short HEAD)
+COMMIT_STATUS ?= $(shell test -n "$(git status --porcelain)")
+BUILD_COMMIT_STATUS ?= $(if $(COMMIT_STATUS),$(BUILD_COMMIT),$(BUILD_COMMIT)-dirty)
```

Comment:
i think there might be something wrong with this if, it appends -dirty when git status is clean

Reply:
With my most recent changes, I switched my approach & following the how the k8s sigs Cluster API Provider for Azure has approached building ldflags (https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/hack/version.sh). I trust the bash `if` more than the make one